### PR TITLE
fix: thorchain LP token withdraws

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

Spotted during final LP testing https://github.com/shapeshift/web/pull/6344#issuecomment-1981969175

- Fixes the calldata for token withdraws to not try and intent a token transfer, which is obviously not valid for withdraws.
- Ensures we send some dust for token withdraws to ensure the withdraw Tx is seen by THOR

## Pull Request Type

- [x] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

N/A

> What protocols, transaction types or contract interactions might be affected by this PR?

## Testing

- Withdraw token Txs fees estimation is happy
- Withdraw token Txs broadcast fine
- Withdraw token Txs actually withdraw

### Engineering

- ^

### Operations

- ^

### Develop

- Sad fees estimation

### This diff

- Happy fees estimation

- Happy Tx

- Happy Tx on the THOR side

### CODE REVIEW

**Comment by gomesalexandre** [comment]
* **#6374** <a href="https://app.graphite.dev/github/pr/shapeshift/web/6374?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `develop`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @gomesalexandre and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: src/pages/ThorChainLP/components/ReusableLpStatus/TransactionRow.tsx [hunk 1]
```diff
@@ -319,21 +319,26 @@ export const TransactionRow: React.FC<TransactionRowProps> = ({
...
-            : // Native EVM assets use the 0 address as the asset address
-              // https://dev.thorchain.org/concepts/sending-transactions.html#admonition-info-1
-              zeroAddress,
+          asset:
+            // The asset param is a directive to initiate a transfer of said asset from the wallet to the contract
+            // which is *not* what we want for withdraws, see
+            // https://www.tdly.co/shared/simulation/6d23d42a-8dd6-4e3e-88a8-62da779a765d
+            isToken(fromAssetId(assetId).assetReference) && isDeposit
+              ? getAddress(fromAssetId(assetId).assetReference)
+              : // Native EVM asset deposits and token withdraws use the 0 address as the asset address
```
Comment (kaladinlight) [inline src/pages/ThorChainLP/components/ReusableLpStatus/TransactionRow.tsx:328]:
```suggestion
              : // Native EVM asset deposits and withdrawals and token withdraws use the 0 address as the asset address
```

File: src/pages/ThorChainLP/components/ReusableLpStatus/TransactionRow.tsx [hunk 2]
```diff
@@ -319,21 +319,26 @@ export const TransactionRow: React.FC<TransactionRowProps> = ({
...
           expiry: BigInt(dayjs().add(15, 'minute').unix()),
         })
 
         return {
           // amountCryptoPrecision is always denominated in fee asset - the only value we can send when calling a contract is native asset value
-          amountCryptoPrecision: isToken(fromAssetId(assetId).assetReference)
-            ? '0'
-            : fromBaseUnit(amountOrDustCryptoBaseUnit, feeAsset.precision),
+          amountCryptoPrecision:
+            isToken(fromAssetId(assetId).assetReference) && isDeposit
```
Comment (kaladinlight) [inline src/pages/ThorChainLP/components/ReusableLpStatus/TransactionRow.tsx:339]: actual tx value should still be 0 for tokens regardless of deposit or withdrawal
```suggestion
            isToken(fromAssetId(assetId).assetReference)
```

File: src/pages/ThorChainLP/components/ReusableLpStatus/TransactionRow.tsx [hunk 3]
```diff
@@ -486,11 +491,15 @@ export const TransactionRow: React.FC<TransactionRowProps> = ({
...
-                : // Native EVM assets use the 0 address as the asset address
-                  // https://dev.thorchain.org/concepts/sending-transactions.html#admonition-info-1
-                  zeroAddress,
+              // The asset param is a directive to initiate a transfer of said asset from the wallet to the contract
+              // which is *not* what we want for withdraws, see
+              // https://www.tdly.co/shared/simulation/6d23d42a-8dd6-4e3e-88a8-62da779a765d
+              asset:
+                isToken(fromAssetId(assetId).assetReference) && isDeposit
+                  ? getAddress(fromAssetId(assetId).assetReference)
+                  : // Native EVM asset deposits and token withdraws use the 0 address as the asset address
```
Comment (kaladinlight) [inline src/pages/ThorChainLP/components/ReusableLpStatus/TransactionRow.tsx:500]:
```suggestion
                  : // Native EVM asset deposits and withdrawals and token withdraws use the 0 address as the asset address
```

**Comment by kaladinlight** [comment]
Good catch on the `depositWithExpiry` amount!

File: src/pages/ThorChainLP/components/ReusableLpStatus/TransactionRow.tsx (hunk 2 above)
Comment (gomesalexandre) [reply src/pages/ThorChainLP/components/ReusableLpStatus/TransactionRow.tsx:339]: That was the actual issue at hand here, broadcasted many many USDC on AVAX withdraws and couldn't get them to be seen. After comparing against TS, noticed they use the exact same dust value as we have defined in constants, adding it made the Txs finally be seen

File: src/pages/ThorChainLP/components/ReusableLpStatus/TransactionRow.tsx [hunk 4]
```diff
@@ -503,9 +512,10 @@ export const TransactionRow: React.FC<TransactionRowProps> = ({
               adapter,
               data,
               // value is always denominated in fee asset - the only value we can send when calling a contract is native asset value
-              value: isToken(fromAssetId(assetId).assetReference)
-                ? '0'
-                : amountOrDustCryptoBaseUnit,
+              value:
+                isToken(fromAssetId(assetId).assetReference) && isDeposit
```
[the remark this answers is not in the document]
Comment (gomesalexandre) [reply src/pages/ThorChainLP/components/ReusableLpStatus/TransactionRow.tsx:516]: Same comment as above

File: src/pages/ThorChainLP/components/ReusableLpStatus/TransactionRow.tsx (hunk 2 above)
Comment (kaladinlight) [reply src/pages/ThorChainLP/components/ReusableLpStatus/TransactionRow.tsx:339]: I see, thought it was related to gas estimation. can we update the comment to give context here then for withdrawal?

Comment (gomesalexandre) [reply src/pages/ThorChainLP/components/ReusableLpStatus/TransactionRow.tsx:339]: Yeah the docs definitely don't seem up-to-date here, they're also not mentioning all assets, only native ones so it's really a guessing game (though savers one are/were a bit more furnished) which makes it so we got it right

Comment (gomesalexandre) [reply src/pages/ThorChainLP/components/ReusableLpStatus/TransactionRow.tsx:340]: > I see, thought it was related to gas estimation. can we update the comment to give context here then for withdrawal?

0e0778e5d63cbcb1822b1d506d06b8601a060f26

File: src/pages/ThorChainLP/components/ReusableLpStatus/TransactionRow.tsx (hunk 3 above)
Comment (gomesalexandre) [reply src/pages/ThorChainLP/components/ReusableLpStatus/TransactionRow.tsx]: 0e0778e5d63cbcb1822b1d506d06b8601a060f26

**Review by kaladinlight — APPROVED** [review]

**Pull request merged**